src/dao/api.dao.ts

```diff
@@ -194,21 +255,73 @@ export class NodeDao {
     });
   };
 
-  getFeedEntries = async (offset: number, limit: number) => {
-    return  await prisma.$queryRaw`
-    SELECT n1.name as name, n1.thumbnail as thumbnail, n1."nodeUri" as link, c.id as claim_id, c.statement as statement, c.stars as stars, c.score as score, c.amt as amt, c."effectiveDate" as effective_date, c."howKnown" as how_known, c.aspect as aspect, c.confidence as confidence, e.label as claim, e2.label as basis, n3.name as source_name, n3.thumbnail as source_thumbnail, n3."nodeUri" as source_link
-    FROM "Node" AS n1
-    INNER JOIN "Edge" AS e ON n1.id = e."startNodeId"
-    INNER JOIN "Node" AS n2 ON e."endNodeId" = n2.id
-    INNER JOIN "Edge" as e2 on n2.id = e2."startNodeId"
-    INNER JOIN "Node" as n3 on e2."endNodeId" = n3.id
-    INNER JOIN "Claim" as c on e."claimId" = c.id
-    WHERE NOT (n1."entType" = 'CLAIM' and e.label = 'source')
-    ORDER BY c."effectiveDate" DESC
-    LIMIT ${limit}
-    OFFSET ${offset}
-  `;
-  }
+  getFeedEntries = async (offset: number, limit: number, search: string) => {
+    const feedEntries = await prisma.$queryRaw<FeedEntry[]>`
+      WITH ranked_claims AS (
+        SELECT
+          c.*,
+          ROW_NUMBER() OVER (PARTITION BY c.id ORDER BY c."effectiveDate" DESC) as rn
+        FROM "Claim" c, "Image" i, "ClaimData" cd
+      )
+      SELECT
+        n1.name as name,
+        n1.thumbnail as thumbnail,
+        n1."nodeUri" as link,
+        n1."descrip" as description,
+        c.id as claim_id,
+        c.statement as statement,
+        c.stars as stars,
+        c.score as score,
+        c.amt as amt,
+        c."effectiveDate" as effective_date,
+        c."howKnown" as how_known,
+        c.aspect as aspect,
+        c.confidence as confidence,
+        e.label as claim,
+        e2.label as basis,
+        n3.name as source_name,
+        n3.thumbnail as source_thumbnail,
+        n3."nodeUri" as source_link,
+        n3."descrip" as source_description,
+        cd.name as claim_name,
+        i.url as image_url,
+        i.digestMultibase as image_digest,
+        i.metadata as image_metadata
+      FROM ranked_claims c
+      INNER JOIN "Edge" AS e ON c.id = e."claimId"
+      INNER JOIN "Node" AS n1 ON e."startNodeId" = n1.id
+      INNER JOIN "Node" AS n2 ON e."endNodeId" = n2.id
+      LEFT JOIN "Edge" as e2 ON n2.id = e2."startNodeId"
+      LEFT JOIN "Node" as n3 ON e2."endNodeId" = n3.id
+      LEFT JOIN "ClaimData" as cd ON c.id = cd."claimId"
+      LEFT JOIN "Image" as i ON c.id = i."claimId"
+      WHERE NOT (n1."entType" = 'CLAIM' AND e.label = 'source')
+      AND (
+        c.statement ILIKE ${`%${search}%`} OR
+        c."sourceURI" ILIKE ${`%${search}%`} OR
```

Comment:
lets remove the extra ones here i think this is going to be slow and we don't need all

Reply:
But these are important, they guarantee the search result includes all the possibilities